drivers/storage/portworx/components_test.go

```diff
@@ -13633,6 +13639,329 @@ func TestDeleteNodePodDisruptionBudget(t *testing.T) {
+	fakeK8sNodes.Items[0].Spec.Unschedulable = true
+	fakeK8sNodes.Items[1].Spec.Unschedulable = true
+	err = testutil.Update(k8sClient, &fakeK8sNodes.Items[0])
+	require.NoError(t, err)
+	err = testutil.Update(k8sClient, &fakeK8sNodes.Items[1])
+	require.NoError(t, err)
+
+	UpgradeNodesResponse := &osdapi.SdkFilterNonOverlappingNodesResponse{
+		NodeIds: []string{"pxnode1", "pxnode2"},
+	}
+	mockNodeServer.EXPECT().FilterNonOverlappingNodes(gomock.Any(), gomock.Any()).Return(UpgradeNodesResponse, nil).Times(1)
+
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	pdbList = &policyv1.PodDisruptionBudgetList{}
+	err = testutil.List(k8sClient, pdbList)
+	require.NoError(t, err)
+	require.Len(t, pdbList.Items, 6)
+	require.Equal(t, "px-node1", pdbList.Items[1].Name)
+	require.Equal(t, 0, pdbList.Items[1].Spec.MinAvailable.IntValue())
+	require.Equal(t, "px-node2", pdbList.Items[2].Name)
+	require.Equal(t, 0, pdbList.Items[2].Spec.MinAvailable.IntValue())
+	require.Equal(t, "px-node3", pdbList.Items[3].Name)
+	require.Equal(t, 1, pdbList.Items[3].Spec.MinAvailable.IntValue())
+
+	// Testcase : When FilterNonOverlappingNodes returns nodes but quorum number of nodes are already upgrading/ ready for upgrade, don't do anything
+	fakeK8sNodes.Items[2].Spec.Unschedulable = true
+	err = testutil.Update(k8sClient, &fakeK8sNodes.Items[2])
+	require.NoError(t, err)
+	UpgradeNodesResponse.NodeIds = []string{"pxnode3"}
+
+	mockNodeServer.EXPECT().FilterNonOverlappingNodes(gomock.Any(), gomock.Any()).Return(UpgradeNodesResponse, nil).Times(1)
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	pdbList = &policyv1.PodDisruptionBudgetList{}
+	err = testutil.List(k8sClient, pdbList)
+	require.NoError(t, err)
+	require.Equal(t, 0, pdbList.Items[1].Spec.MinAvailable.IntValue())
+	require.Equal(t, 0, pdbList.Items[2].Spec.MinAvailable.IntValue())
+	require.Equal(t, 1, pdbList.Items[3].Spec.MinAvailable.IntValue())
+
+	// Testcase : When user provides minAvailable annotation, use that instead of portworx quorum
+	minAvailable := intstr.FromInt(1)
+	for i := 1; i < 4; i++ {
+		pdbList.Items[i].Spec.MinAvailable = &minAvailable
+		err = testutil.Update(k8sClient, &pdbList.Items[i])
+		require.NoError(t, err)
+	}
+	UpgradeNodesResponse.NodeIds = []string{"pxnode1", "pxnode2"}
+	mockNodeServer.EXPECT().FilterNonOverlappingNodes(gomock.Any(), gomock.Any()).Return(UpgradeNodesResponse, nil).Times(1)
+
+	cluster.Annotations = map[string]string{
+		pxutil.AnnotationStoragePodDisruptionBudget: "4",
+	}
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	pdbList = &policyv1.PodDisruptionBudgetList{}
+	err = testutil.List(k8sClient, pdbList)
+	require.NoError(t, err)
+	require.Equal(t, 0, pdbList.Items[1].Spec.MinAvailable.IntValue())
+	require.Equal(t, 1, pdbList.Items[2].Spec.MinAvailable.IntValue())
+	require.Equal(t, 1, pdbList.Items[3].Spec.MinAvailable.IntValue())
+
+	// Testcase: When minAvailable is invalid, use portworx quorum
```

Comment:
This should be divided in two cases. Here the modified PDB count is not equal to quorum just because that is the quorum number. It is also because the PX API response also returns only 2 nodes. To really test quorum, try returning 4 nodes from the API response and we should test that only 2 PDBs are modified in that case.